test/functional/rpc_whitelist.py

```diff
@@ -92,5 +90,56 @@ def run_test(self):
         self.log.info("Strange test 5")
         assert_equal(200, rpccall(self.nodes[0], self.strange_users[4], "getblockcount").status)
 
+        self.test_users_permissions()
+        self.test_rpcwhitelistdefault_0_no_permissions()
+
+        # Replace file configurations
+        self.nodes[0].replace_in_config([("rpcwhitelistdefault=0", "rpcwhitelistdefault=1")])
+        with open(self.nodes[0].datadir_path / "bitcoin.conf", 'a', encoding='utf8') as f:
+            f.write("rpcwhitelist=__cookie__:getblockcount,getblockchaininfo,getmempoolinfo,stop\n")
+        self.restart_node(0)
+
+        # Test rpcwhitelistdefault=1
+        self.test_users_permissions()
+        self.test_rpcwhitelistdefault_1_no_permissions()
+
+    def test_users_permissions(self):
+        """
+        * Permissions:
+            (user1): getbestblockhash,getblockcount
+            (user2): getblockcount
+        Expected result:  * users can only access whitelisted methods
+        """
+        for user in self.users:
+            permissions = get_permissions(user[2])
+            for permission in permissions:
+                self.log.info("[" + user[0] + "]: Testing whitelisted user permission (" + permission + ")")
```

Comment:
nit: You can avoid the verbose concatenations by using python formatted string, here and other places
```suggestion
                self.log.info(f"[{user[0]}]: Testing whitelisted user permission ({permission})")
```